src/native/clr/host/bridge-processing.cc

```
@@ -0,0 +1,307 @@
+	}
+}
+
+void BridgeProcessing::clear_references_if_needed (JniObjectReferenceControlBlock *control_block) noexcept
+{
+	abort_if_invalid_pointer_argument (control_block, "control_block");
+	abort_unless (control_block->handle != nullptr, "Control block handle must not be null");
+	abort_unless (control_block->handle_type == JNIGlobalRefType, "Control block handle type must be global reference");
+
+	if (control_block->refs_added == 0) {
+		return;
+	}
+
+	// Clear references from the object
+	jclass java_class = env->GetObjectClass (control_block->handle);
+	jmethodID clear_method_id = env->GetMethodID (java_class, "monodroidClearReferences", "()V");
+	env->DeleteLocalRef (java_class); // Clean up the local reference to the class
+
+	if (clear_method_id != nullptr) [[likely]] {
+		env->CallVoidMethod (control_block->handle, clear_method_id);
+		control_block->refs_added = 0;
+	} else {
+		log_error (LOG_DEFAULT, "Failed to find monodroidClearReferences method");
+		env->ExceptionClear ();
+#if DEBUG
+		if (Logger::gc_spew_enabled ()) {
+			char *class_name = Host::get_java_class_name_for_TypeManager (java_class);
+			log_error (LOG_GC, "Missing monodroidClearReferences method for object of class {}", class_name);
+			free (class_name);
+		}
+#endif
+	}
+}
+
+void BridgeProcessing::take_global_ref (HandleContext *context) noexcept
+{
+	abort_if_invalid_pointer_argument (context, "context");
+	abort_unless (context->control_block != nullptr, "Control block must not be null"); // TODO: this sometimes throws -- looks like a race condition?
+	if (context->control_block->handle_type != JNIWeakGlobalRefType) [[unlikely]] {
+		log_error (LOG_DEFAULT, "Expected weak global reference type for handle, but got {} - handle: {:#x}", context->control_block->handle_type, reinterpret_cast<intptr_t> (context->control_block->handle));
+		return;
+	}
+	abort_unless (context->control_block->handle_type == JNIWeakGlobalRefType, "Expected weak global reference type for handle");
```

Remove this, as the above `if` handles the situation already. If we really want to abort, use `Helper::abort_application` in the `if` block instead of `log_error`